src/pages/Main.tsx

```diff
@@ -1,7 +1,58 @@
-import { Button } from "@deskpro/app-sdk";
+/* eslint-disable react-hooks/exhaustive-deps */
+import { TextArea, GetStateResponse } from "@deskpro/app-sdk";
+import { Stack, Button, H1, useDeskproAppClient } from "@deskpro/app-sdk";
+import { useState, useEffect } from "react";
+
+import useDebounce from "../utils/debounce";
 
 export const Main = () => {
-    return (
-        <Button text="Greet" onClick={() => alert("Hello")} />
-    );
+  const { client } = useDeskproAppClient();
+
+  const [text, setText] = useState<string>("");
+  const debouncedText = useDebounce(text, 200);
+
+  useEffect(() => {
```

Comment:
We've got a new way of doing this using a special hook. It basically does what you're doing here but encapsulates the function in a test for an initialized Deskpro apps client. You can see an example of it here: https://support.deskpro.com/hr-HR/guides/developers/how-to-use-state-to-store-notes#anchor-1

So this code would change to something like:

```typescript
useInitialisedDeskproAppClient((client) => {
  client.setState<string>("scratchpad", debouncedText);
}, [debouncedText]);
```